### PR TITLE
Task #163318 chore: Certificate page detail view

### PR DESCRIPTION
**Pull request by tushar33**
1. On page load render image.
2. Save image on server.
3. Add a social sharing backend config as per used in our products.
4. Add multiple selection config for social sharing option

### CODE REVIEW

File: src/components/com_tjcertificate/media/js/tjCertificate.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
+jQuery(document).ready(function() {
```
Comment (praneettekdi) [inline src/components/com_tjcertificate/media/js/tjCertificate.js:1]: Can you write JS using Objects like other JS file in the folder

File: src/components/com_tjcertificate/administrator/controllers/certificate.php [hunk 2]
```diff
@@ -56,4 +56,40 @@ public function download()
...
+		$canvasOutput = str_replace('data:image/png;base64,', '', $canvasOutput);
+
+		// Replace all spaces with plus sign (helpful for larger images)
+		$canvasOutput = str_replace(" ", "+", $canvasOutput);
+		$canvasOutput = base64_decode($canvasOutput);
+		$filename = $certificateId . '.png';
+
+		if (!JFolder::exists(JPATH_SITE . '/media/com_tjcertificate/certificates/'))
+		{
+			JFolder::create(JPATH_SITE . '/media/com_tjcertificate/certificates', 0777);
```
Comment (praneettekdi) [inline src/components/com_tjcertificate/administrator/controllers/certificate.php:82]: can you try with 655

File: src/components/com_tjcertificate/site/languages/en-GB/en-GB.com_tjcertificate.ini [hunk 3]
```diff
@@ -52,3 +52,5 @@ COM_TJCERTIFICATE_CERTIFICATE_TEMPLATE_FIELD_PRIVATE="Private"
 COM_TJCERTIFICATE_CERTIFICATE_TEMPLATE_FIELD_PUBLIC="Public"
 COM_TJCERTIFICATE_CERTIFICATE_TEMPLATE_ENTER_CERTIFICATE_ID="Enter Certificate Id"
 
+;__DEPLOY_VERSION__
+COM_TJCERTIFICATE_CERTIFICATE_DOWNLOAD="Download"
```
Comment (praneettekdi) [inline src/components/com_tjcertificate/site/languages/en-GB/en-GB.com_tjcertificate.ini:56]: Add a new line at the end

File: src/components/com_tjcertificate/media/js/tjCertificate.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
+jQuery(document).ready(function() {
+    jQuery('#btn-Convert-Html2Image').click(function() {
```
Comment (praneettekdi) [inline src/components/com_tjcertificate/media/js/tjCertificate.js:2]: Rename JS file to certificateImage.js

File: src/components/com_tjcertificate/site/views/certificate/tmpl/default.php [hunk 5]
```diff
@@ -14,15 +14,22 @@
 use Joomla\CMS\Factory;
 use Joomla\CMS\Router\Route;
 use Joomla\CMS\Language\Text;
+use Joomla\CMS\Uri\Uri;
+use Joomla\CMS\HTML\HTMLHelper;
+
+HTMLHelper::StyleSheet('media/com_tjcertificate/vendors/font-awesome-4.1.0/css/font-awesome.min.css');
+HTMLHelper::script('media/com_tjcertificate/vendors/html2canvas/js/html2canvas.js');
+HTMLHelper::script('media/com_tjcertificate/js/tjCertificate.js');
```
Comment (praneettekdi) [inline src/components/com_tjcertificate/site/views/certificate/tmpl/default.php:23]: Load Media JS files like

```php
$options['relative'] = true;
HTMLHelper::script('com_tjcertificate/tjCertificate.js', $options);
```

File: src/components/com_tjcertificate/administrator/controllers/certificate.php (hunk 2 above)
Comment (tushar33) [reply src/components/com_tjcertificate/administrator/controllers/certificate.php:82]: not working

File: src/components/com_tjcertificate/site/languages/en-GB/en-GB.com_tjcertificate.ini (hunk 3 above)
Comment (tushar33) [reply src/components/com_tjcertificate/site/languages/en-GB/en-GB.com_tjcertificate.ini:56]: done

File: src/components/com_tjcertificate/media/js/tjCertificate.js (hunk 4 above)
Comment (tushar33) [reply src/components/com_tjcertificate/media/js/tjCertificate.js:2]: done

File: src/components/com_tjcertificate/administrator/controllers/certificate.php (hunk 2 above)
Comment (praneettekdi) [reply src/components/com_tjcertificate/administrator/controllers/certificate.php:82]: ok. 775?

Comment (tushar33) [reply src/components/com_tjcertificate/administrator/controllers/certificate.php:82]: yes working

Comment (tushar33) [reply src/components/com_tjcertificate/administrator/controllers/certificate.php:82]: Done

File: src/components/com_tjcertificate/media/js/tjCertificate.js (hunk 1 above)
Comment (tushar33) [reply src/components/com_tjcertificate/media/js/tjCertificate.js]: done

File: src/components/com_tjcertificate/site/languages/en-GB/en-GB.com_tjcertificate.ini (hunk 3 above)
Comment (tushar33) [reply src/components/com_tjcertificate/site/languages/en-GB/en-GB.com_tjcertificate.ini]: done

File: src/components/com_tjcertificate/site/views/certificate/tmpl/default.php (hunk 5 above)
Comment (tushar33) [reply src/components/com_tjcertificate/site/views/certificate/tmpl/default.php:24]: loaded for certificateImage.js but vendor js is not loading with adding "$options"

**Pull request closed without merging**